src/arkode/arkode_io.c

```diff
@@ -1640,11 +2386,47 @@ int arkGetCurrentTime(void* arkode_mem, sunrealtype* tcur)
 }
 
 /*---------------------------------------------------------------
-  arkGetTolScaleFactor:
+  ARKodeGetCurrentGamma: Returns the current value of gamma
+  ---------------------------------------------------------------*/
+int ARKodeGetCurrentGamma(void* arkode_mem, sunrealtype* gamma)
+{
+  ARKodeMem ark_mem;
+  if (arkode_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  ark_mem = (ARKodeMem)arkode_mem;
+
+  /* Guard against use for time steppers that do not need an algebraic solver */
+  if (!ark_mem->step_supports_algebraic)
+  {
+    arkProcessError(ark_mem, ARK_STEPPER_UNSUPPORTED, __LINE__, __func__,
+                    __FILE__, "time-stepping module does not require an algebraic solver");
+    return (ARK_STEPPER_UNSUPPORTED);
+  }
+
+  /* Call stepper routine to compute the state (if provided) */
+  if (ark_mem->step_getcurrentgamma)
+  {
+    return (ark_mem->step_getcurrentgamma(ark_mem, gamma));
+  }
+  else
+  {
+    arkProcessError(ark_mem, ARK_STEPPER_UNSUPPORTED, __LINE__, __func__,
+                    __FILE__,
+                    "time-stepping module does not support this function");
+    return (ARK_STEPPER_UNSUPPORTED);
+  }
+}
+
+/*---------------------------------------------------------------
+  ARKodeGetTolScaleFactor:
 
   Returns a suggested factor for scaling tolerances
   ---------------------------------------------------------------*/
-int arkGetTolScaleFactor(void* arkode_mem, sunrealtype* tolsfact)
+int ARKodeGetTolScaleFactor(void* arkode_mem, sunrealtype* tolsfact)
```

Comment:
The value of `tolsf` is independent of the estimated temporal error, it really just asks whether the tolerances are incompatible with the magnitudes present in the solution vector.  Since we also use tolerances for implicit solvers, we could only remove this for SPRKStep (or we could check whether one of `supports_algebraic` or `supports_adaptivity` is enabled).